### PR TITLE
Yet another 3 fix

### LINKED ISSUE #162: Potential Bug
- [ ] I have marked all applicable categories:
    + [√ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [ √] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [ √] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ √] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, sys
  print(tianshou.__version__, torch.__version__, sys.version, sys.platform)
  # 0.2.4 1.4.0 3.7.6 | packaged by conda-forge | (default, Jun  1 2020, 18:57:50) [GCC 7.5.0] linux
  ```
Hi, first thanks for the excellent work! 

One thing I found may has potential bug. 
When I was using a2c policy, I found that you calculate [a_loss](https://github.com/thu-ml/tianshou/blob/dev/tianshou/policy/modelfree/a2c.py#L111) in this way, which may not work if the action has multiple dimension, say if the shape of output of dist.log_prob(a) is [bsz, n] while the shape of (r-v) is [bsz, 1] or [bsz] and neither of them can do dot product with the dis.log_prob(a). In my opinion, it would be better to use dist.log_prob(a).transpose(0,1) so that it will do multiplication among the same dimension.

### PR DESCRIPTION
**Pull request by Trinkle23897**
- [x] I have marked all applicable categories:
    + [x] algorithm implementation fix
    + [x] documentation modification

1. DQN learn should keep eps=0
2. Add a warning of env.seed in VecEnv
3. fix #162 of multi-dim action

### CODE REVIEW

**Comment by Trinkle23897** [comment]
It's no difference with the previous version because `self(batch).act` is not used in `learn()`, since `eps` only changes `act`.
But pointing it out is much better.

File: tianshou/utils/net/common.py [hunk 1]
```diff
@@ -36,7 +36,10 @@ def __init__(self, layer_num, state_shape, action_shape=0, device='cpu',
     def forward(self, s, state=None, info={}):
         """s -> flatten -> logits"""
         s = to_torch(s, device=self.device, dtype=torch.float32)
-        s = s.flatten(1)
+        if len(s.shape) > 1:
```
Comment (duburcqa) [inline tianshou/utils/net/common.py]: it is quite ugly x) what is the shape of s ?

Comment (Trinkle23897) [reply tianshou/utils/net/common.py]: It is not determined, either [bsz] or [bsz, ?] or [bsz, ?, ?] or [bsz, ?, ?, ?]. I want to change the shape as (bsz, -1), so it's better to use reshape

**Review by youkaichao — APPROVED** [review]

**Comment by youkaichao** [comment]
Reshape is good, and would be better if you can avoid shape [bsz] by all means (reward has this shape) because of the [confusing behavior](https://stackoverflow.com/questions/40866598/broadcasting-n-and-with-n-1-arrays-in-numpy).

File: tianshou/utils/net/common.py (hunk 1 above)
Comment (duburcqa) [reply tianshou/utils/net/common.py]: ok so why not `s.flatten(1, -1)` ?

**Comment by duburcqa** [comment]
> Reshape is good, and would be better if you can avoid shape [bsz] by all means (reward has this shape) because of the confusing behavior.

I don't agree, usually I prefer to use `.flatten(1, -1)` since it makes it clear that you want to flatten but preserve batch processing.

**Comment by Trinkle23897** [comment]
> > Reshape is good, and would be better if you can avoid shape [bsz] by all means (reward has this shape) because of the confusing behavior.
> 
> I don't agree, usually I prefer to use `.flatten(1, -1)` since it makes it clear that you want to flatten but preserve batch processing.

But 1-dim tensor cannot apply flatten(1):

```python
In [6]: b=torch.rand(3)

In [7]: b.flatten(1,-1)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-7-0fd10ca82bb9> in <module>
----> 1 b.flatten(1,-1)

IndexError: Dimension out of range (expected to be in range of [-1, 0], but got 1)
```

**Comment by duburcqa** [comment]
> But 1-dim tensor cannot apply flatten(1):

OK so your example before edit "`[bsz, ?]` or `[bsz, ?, ?]`" was wrong. So yes if `[bsz]` can happen (which I highly doubt, since it means 0-dim input without batch-processing) then `reshape` must be used.

**Comment by youkaichao** [comment]
>  if [bsz] can happen (which I highly doubt, since it means 0-dim input without batch-processing)

reward is the case.

**Comment by duburcqa** [comment]
> reward is the case.

The reward is used as input of its own neural network ? In which algorithm ?

**Comment by youkaichao** [comment]
> The reward is used as input of its own neural network

I'm not talking about using reward as input, though. My suggestion is to reshape reward to be [bsz, 1] in ``Policy.learn`` :)

**Comment by Trinkle23897** [comment]
> > The reward is used as input of its own neural network
> 
> I'm not talking about using reward as input, though. My suggestion is to reshape reward to be [bsz, 1] in `Policy.learn` :)

I don't agree to reshape this as [bsz, 1] since it had already made a bug previously.

**Comment by duburcqa** [comment]
> I'm not talking about using reward as input, though. My suggestion is to reshape reward to be [bsz, 1] in Policy.learn :)

We are talking about [this](https://github.com/thu-ml/tianshou/pull/160#discussion_r459922060) initially.

**Comment by Trinkle23897** [comment]
Okay. This is because I write a small test with MyTestEnv under the `test/base/` folder. I found that the obs in this env cannot be processed through the current network (because its shape [bsz]), so I modify this line of code.
I know it rarely happens :)

**Review by duburcqa — APPROVED** [review]

**Pull request merged**